src/System.Net.Security/src/System/Net/_SSPIWrapper.cs

```diff
@@ -0,0 +1,873 @@
+            {
+                lock (SecModule)
+                {
+                    if (SecModule.SecurityPackages == null)
+                    {
+                        int moduleCount = 0;
+                        SafeFreeContextBuffer arrayBaseHandle = null;
+                        try
+                        {
+                            int errorCode = SecModule.EnumerateSecurityPackages(out moduleCount, out arrayBaseHandle);
+                            GlobalLog.Print("SSPIWrapper::arrayBase: " + (arrayBaseHandle.DangerousGetHandle().ToString("x")));
+                            if (errorCode != 0)
+                            {
+                                throw new Win32Exception(errorCode);
+                            }
+                            SecurityPackageInfoClass[] securityPackages = new SecurityPackageInfoClass[moduleCount];
+                            if (Logging.On) Logging.PrintInfo(Logging.Web, SR.net_log_sspi_enumerating_security_packages);
+                            int i;
+                            for (i = 0; i < moduleCount; i++)
+                            {
+                                securityPackages[i] = new SecurityPackageInfoClass(arrayBaseHandle, i);
+                                if (Logging.On) Logging.PrintInfo(Logging.Web, "    " + securityPackages[i].Name);
+                            }
+                            SecModule.SecurityPackages = securityPackages;
+                        }
+                        finally
+                        {
+                            if (arrayBaseHandle != null)
+                            {
+                                arrayBaseHandle.Dispose();
+                            }
+                        }
+                    }
+                }
+            }
+            GlobalLog.Leave("EnumerateSecurityPackages");
+            return SecModule.SecurityPackages;
+        }
+
+        internal static SecurityPackageInfoClass GetVerifyPackageInfo(SSPIInterface secModule, string packageName)
+        {
+            return GetVerifyPackageInfo(secModule, packageName, false);
+        }
+
+        internal static SecurityPackageInfoClass GetVerifyPackageInfo(SSPIInterface secModule, string packageName, bool throwIfMissing)
+        {
+            SecurityPackageInfoClass[] supportedSecurityPackages = EnumerateSecurityPackages(secModule);
+            if (supportedSecurityPackages != null)
+            {
+                for (int i = 0; i < supportedSecurityPackages.Length; i++)
+                {
+                    if (string.Compare(supportedSecurityPackages[i].Name, packageName, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        return supportedSecurityPackages[i];
+                    }
+                }
+            }
+
+            if (Logging.On) Logging.PrintInfo(Logging.Web, SR.Format(SR.net_log_sspi_security_package_not_found, packageName));
```

Comment:
Actually, it's rule 1) "... A single line statement block can go without braces but the block <em>must be properly indented on its own line</em> and it must not be nested in other statement blocks that use braces (See issue 381 for examples)." (emphasis mine).

I used to be in the "never have unbraced ifs" camp, now I'm in the "they're only allowed in preconditions" camp (which may be made up of just me).

I'm not necessarily advocating for or against the rule here; just pointing out that we have one.